### DISCUSSION TITLE
Moving code from Javascript to Java library

### DISCUSSION THREAD

**Discussion by mcswaip1** [nextgenhealthcare/connect#4830] [Q&A] [answered]
I have several Javascript methods that all of my channels need access to; currently I have them in the GlobalMap, but I would really like to get them into a JAR instead.

Most of these access a database to query information, but I have not found any examples of creating a custom Java library that accesses a db via the `com.mirth.connect.server.userutil.DatabaseConnectionFactory`.

Can you point me in the right direction?
Thanks

**Comment by pacmano1** [comment]
Why not use code templates for this, that is what they are for.  Just curious really.

**Reply by mcswaip1** [reply]
I looked into code templates, but they seem to need updating every channel that uses the code template when a change is made. This would mean that 5-8 channels would need to be redeployed if a code change happens for a single issue. This means that we would need to package all channels with each release which seems excessive. Using a Java Library would get around this. Unless I am mistaken on how code templates work.

**Reply by pacmano1** [reply]
I appreciate your point.

If you edit a code template, you do need to re-deploy the channel, this a trade off, you are correct.    In practice for us that isn't actually a frequent occurrence since global scripts are well tested and few in number. 

For example  DB "setup cached connection" templates, I don't think that code has changed in years.   And given most stuff we write is inherently multi-tenant, sharing SQL statements across channels is rare, and in fact nearly all calls are SPs.  

(just noting that stuff for other readers)

Hopefully someone will chime in on your original question.

**Reply by tonygermano** [reply]
I'm not sure if you are both using "redeploy" in the same sense. If you update a code template, you don't need to update the channels. You only need to redeploy them within mirth to pick up the code template changes. If you are packaging and importing your changes, only the code template would need to be imported, not all of the channels to which it is assigned.

**Reply by pacmano1** [reply]
Good point - I meant re-deploy the channel to pick up the code template updates.

**Comment by tonygermano** [comment, marked as answer]
While I think @pacmano1 's answer to use code templates is the correct one, why are you having issues using `com.mirth.connect.server.userutil.DatabaseConnectionFactory`? You should be able to call it like any other java library.
